plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -0,0 +1,159 @@
+
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.services.dynamodb.model.DeleteItemRequest;
+import software.amazon.awssdk.services.dynamodb.model.GetItemRequest;
+import software.amazon.awssdk.services.dynamodb.model.GetItemResponse;
+import software.amazon.awssdk.services.dynamodb.model.PutItemRequest;
+
+/**
+ * DDB implementation of {@link SdkClient}. DDB table name will be mapped to index name.
+ *
+ */
+@AllArgsConstructor
+@Log4j2
+public class DDBOpenSearchClient implements SdkClient {
+
+    private static final String DEFAULT_TENANT = "DEFAULT_TENANT";
+
+    private static final String HASH_KEY = "tenant_id";
+    private static final String RANGE_KEY = "id";
+    private static final String SOURCE = "source";
+
+    private DynamoDbClient dynamoDbClient;
+
+    /**
+     * DDB implementation to write data objects to DDB table. Tenant ID will be used as hash key and document ID will
+     * be used as range key. If tenant ID is not defined a default tenant ID will be used. If document ID is not defined
+     * a random UUID will be generated. Data object will be written as a nested DDB attribute.
+     *
+     */
+    @Override
+    public CompletionStage<PutDataObjectResponse> putDataObjectAsync(PutDataObjectRequest request, Executor executor) {
+        final String id = request.id() != null ? request.id() : UUID.randomUUID().toString();
+        final String tenantId = request.tenantId() != null ? request.tenantId() : DEFAULT_TENANT;
+        final String tableName = getTableName(request.index());
+        return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<PutDataObjectResponse>) () -> {
+            try (XContentBuilder sourceBuilder = XContentFactory.jsonBuilder()) {
+                XContentBuilder builder = request.dataObject().toXContent(sourceBuilder, ToXContent.EMPTY_PARAMS);
+                String source = builder.toString();
+
+                final Map<String, AttributeValue> item = Map
+                    .ofEntries(
+                        Map.entry(HASH_KEY, AttributeValue.builder().s(tenantId).build()),
+                        Map.entry(RANGE_KEY, AttributeValue.builder().s(id).build()),
+                        Map.entry(SOURCE, AttributeValue.builder().s(source).build())
+                    );
+                final PutItemRequest putItemRequest = PutItemRequest.builder().tableName(tableName).item(item).build();
+
+                dynamoDbClient.putItem(putItemRequest);
+                return new PutDataObjectResponse.Builder().id(id).created(true).build();
+            } catch (Exception e) {
+                log.error("Exception while inserting data into DDB: " + e.getMessage(), e);
+                throw new OpenSearchException(e);
+            }
```

Comment:
Probably want better exception handling than this, see patterns in https://github.com/opensearch-project/ml-commons/pull/2507

Better yet, probably want to merge that and rebase this PR CC: @dhrubo-os